Author: Eli Zaretskii    eliz gnu org
Description: Resurrect highlighting of repeated words by Flyspell Mode
Upstream: yes 
Upstream-Commit: 33b779a11fb6785944383aeeae44f77cb580ee37 (Wed Jul 29 20:52:23 2015 +0300)
Upstream-Bug: http://debbugs.gnu.org/21157

    * lisp/textmodes/flyspell.el (flyspell-word): Leave some non-word
    characters between point and the doublon candidate, so that
    flyspell-word-search-backward finds it.  (Bug#21157)

But make sure its calculation does not change focus.

--- a/support/emacsen/flyspell.el
+++ b/support/emacsen/flyspell.el
@@ -1115,7 +1115,8 @@
 		   (let* ((bound
 			   (- start
 			      (- end start)
-			      (- (skip-chars-backward " \t\n\f"))))
+			      (- (save-excursion
+                                   (skip-chars-backward " \t\n\f")))))
 			  (p (when (>= bound (point-min))
 			       (flyspell-word-search-backward word bound t))))
 		     (and p (/= p start)))))
